server.js

```diff
@@ -1632,6 +1625,64 @@ cache(function (data, match, sendBadge, request) {
   });
 }));
 
+// npm weekly download integration.
+camp.route(/^\/npm\/dw\/(.*)\.(svg|png|gif|jpg|json)$/,
+cache(function (data, match, sendBadge, request) {
+  var pkg = encodeURIComponent(match[1]);  // eg, "express" or "@user/express"
+  var format = match[2];
+  var apiUrl = 'https://api.npmjs.org/downloads/point/last-week/' + pkg;
+  var badgeData = getBadgeData('downloads', data);
+  request(apiUrl, function (err, res, buffer) {
+    if (err != null) {
+      badgeData.text[1] = 'inaccessible';
+      sendBadge(format, badgeData);
+      return;
+    }
+    try {
+      var weekly = JSON.parse(buffer).downloads || 0;
+      badgeData.text[1] = metric(weekly) + '/week';
+      if (weekly === 0) {
+        badgeData.colorscheme = 'red';
+      } else {
+        badgeData.colorscheme = 'brightgreen';
+      }
+      sendBadge(format, badgeData);
+    } catch(e) {
+      badgeData.text[1] = 'invalid';
+      sendBadge(format, badgeData);
+    }
+  });
+}));
+
+// npm yearly download integration
+camp.route(/^\/npm\/dy\/(.*)\.(svg|png|gif|jpg|json)$/,
+cache(function (data, match, sendBadge, request) {
+  var pkg = encodeURIComponent(match[1]);  // eg, "express" or "@user/express"
+  var format = match[2];
+  var apiUrl = 'https://api.npmjs.org/downloads/point/last-year/' + pkg;
+  var badgeData = getBadgeData('downloads', data);
+  request(apiUrl, function (err, res, buffer) {
+    if (err != null) {
+      badgeData.text[1] = 'inaccessible';
+      sendBadge(format, badgeData);
+      return;
+    }
+    try {
+      var yearly = JSON.parse(buffer).downloads || 0;
+      badgeData.text[1] = metric(yearly) + '/year';
+      if (yearly === 0) {
+        badgeData.colorscheme = 'red';
+      } else {
+        badgeData.colorscheme = 'brightgreen';
+      }
+      sendBadge(format, badgeData);
+    } catch(e) {
+      badgeData.text[1] = 'invalid';
+      sendBadge(format, badgeData);
+    }
+  });
+}));
```

Comment:
The only things that differ between the three badges seem to be:
 - Badge URL (`dw`, `dm` , `dy`)
 - NPM API URL (`last-week`, `last-month`, `last-year`)
 - Suffix on the badge text (`/week`, `/month`, `/year`)

Everything else is the same. Given that, here's one idea (untested but should work):
```js
function mapNpmDownloads(urlComponent, apiUri, badgeSuffix) {
  camp.route(new RegExp('^\/npm\/' + urlComponent + '\/(.*)\.(svg|png|gif|jpg|json)$'),
  cache(function (data, match, sendBadge, request) {
    // .....
  });
}

mapNpmDownloads('dm', 'last-month', '/month');
mapNpmDownloads('dw', 'last-week', '/week');
mapNpmDownloads('dy', 'last-year', '/year');
```